### PR TITLE
Add missing basket_addition call.  Handle call to post_checkout with no user=

### PR DESCRIPTION
**Pull request by PaulWattenberger**
Add missing basket_addition call.  
Handle signal send to post_checkout with no user= or request=

### CODE REVIEW

**Comment by PaulWattenberger** [comment]
@clintonb Clinton, can you review.  When I ran on stage, i discovered that we aren't calling the basket_addition oscar signal so I had to add a call

File: ecommerce/sailthru/signals.py [hunk 1]
```diff
@@ -44,16 +45,19 @@ def process_checkout_complete(sender, order=None, request=None, user=None, **kwa
         # figure out course url
         course_url = _build_course_url(course_id)
 
+        # get current request
+        request = threadlocals.get_current_request()
```
Comment (clintonb) [inline ecommerce/sailthru/signals.py]: `threadlocals` is evil and I wish it never existed. Why is it being used here?

Comment (PaulWattenberger) [inline ecommerce/sailthru/signals.py]: Copied from https://github.com/edx/ecommerce/blob/e61b0c238cdadaa13ad775b77cb9ce4da581bb85/ecommerce/extensions/checkout/signals.py

Actually, the request is supposed to be passed in the post_checkout signal call, but a couple of the calls we added, don't include it and I didn't want to change them if I didn't have to...

Comment (clintonb) [inline ecommerce/sailthru/signals.py]: Let's do this the right way, without `threadlocals`.

**Comment by PaulWattenberger** [comment]
@clintonb

File: ecommerce/sailthru/signals.py [hunk 2]
```diff
@@ -27,6 +27,12 @@ def process_checkout_complete(sender, order=None, request=None, user=None, **kwa
     if not waffle.switch_is_active('sailthru_enable'):
         return
 
+    site_code = None
+    message_id = None
+    if request:
+        site_code = request.site.siteconfiguration.partner.short_code
```
Comment (clintonb) [inline ecommerce/sailthru/signals.py]: Use `order.site` instead of `request.site`.

File: ecommerce/sailthru/signals.py [hunk 3]
```diff
@@ -17,7 +17,7 @@
 
 @receiver(post_checkout)
 @silence_exceptions("Failed to call Sailthru upon order completion.")
-def process_checkout_complete(sender, order=None, request=None, user=None, **kwargs):  # pylint: disable=unused-argument
+def process_checkout_complete(sender, order=None, request=None, **kwargs):  # pylint: disable=unused-argument
```
Comment (clintonb) [inline ecommerce/sailthru/signals.py]:
``` python
post_checkout = Signal(providing_args=["order", "user", "request", "response"])
```

This signal should list each of the above args as kwargs.

**Comment by clintonb** [comment]
👍 (once comments addressed, commits squashed). Remember to delete your branch after merging.

**Pull request merged**